crates/swift-integration-tests/src/result.rs

```diff
@@ -68,6 +77,24 @@ fn rust_func_takes_result_opaque_swift(
     }
 }
 
+fn rust_func_return_result_null_opaque_rust(succeed: bool) -> Result<(), ResultTestOpaqueRustType> {
+    if succeed {
+        Ok(())
+    } else {
+        Err(ResultTestOpaqueRustType { val: 222 })
+    }
+}
+
+fn rust_func_return_result_unit_struct_opaque_rust(
+    succeed: bool,
+) -> Result<ffi::UnitStruct, ResultTestOpaqueRustType> {
+    if succeed {
+        Ok(ffi::UnitStruct {})
```

Comment:
Does this compile? Seems like it should be `ffi::UnitStruct` without the braces

Reply:
Huh - I didn't know you could do that. Changed, because why not, but yes, this does compile.